packages/core/admin/admin/src/pages/ProfilePage.tsx

```diff
@@ -0,0 +1,710 @@
+
+import {
+  Box,
+  Button,
+  ContentLayout,
+  Flex,
+  HeaderLayout,
+  Main,
+  useNotifyAT,
+  Grid,
+  GridItem,
+  Typography,
+  SingleSelect,
+  SingleSelectOption,
+  TextInput,
+  FieldAction,
+} from '@strapi/design-system';
+import {
+  auth,
+  Form,
+  GenericInput,
+  GenericInputProps,
+  LoadingIndicatorPage,
+  pxToRem,
+  useAppInfo,
+  useFetchClient,
+  useFocusWhenNavigate,
+  useNotification,
+  useOverlayBlocker,
+  useTracking,
+} from '@strapi/helper-plugin';
+import { Check, Eye, EyeStriked } from '@strapi/icons';
+import { AxiosError } from 'axios';
+import { Formik, FormikHelpers } from 'formik';
+import upperFirst from 'lodash/upperFirst';
+import { Helmet } from 'react-helmet';
+import { useIntl } from 'react-intl';
+import { useMutation, useQuery } from 'react-query';
+import styled from 'styled-components';
+import * as yup from 'yup';
+
+import { useLocales } from '../components/LanguageProvider';
+import { useThemeToggle, ThemeName, ThemeToggleContextContextValue } from '../contexts/themeToggle';
+import { getFullName } from '../utils/getFullName';
+
+// @ts-expect-error – no types available
+import { profileValidation } from './SettingsPage/pages/Users/utils/validations/users';
+
+import type { GetMe, UpdateMe } from '../../../shared/contracts/users';
+
+const schema = yup.object().shape(profileValidation);
+
+/* -------------------------------------------------------------------------------------------------
+ * ProfilePage
+ * -----------------------------------------------------------------------------------------------*/
+
+const ProfilePage = () => {
+  const { changeLocale, localeNames } = useLocales();
+  const { setUserDisplayName } = useAppInfo();
+  const { formatMessage } = useIntl();
+  const { trackUsage } = useTracking();
+  const toggleNotification = useNotification();
+  const { lockApp, unlockApp } = useOverlayBlocker();
+  const { notifyStatus } = useNotifyAT();
+  const { currentTheme, themes: allApplicationThemes, onChangeTheme } = useThemeToggle();
+  const { get, put } = useFetchClient();
+
+  useFocusWhenNavigate();
+
+  const {
+    isLoading: isLoadingUser,
+    data,
+    refetch,
+  } = useQuery(
+    'user',
+    async () => {
+      const { data } = await get<GetMe.Response>('/admin/users/me');
+
+      return data.data;
+    },
+    {
+      onSuccess() {
+        notifyStatus(
+          formatMessage({
+            id: 'Settings.profile.form.notify.data.loaded',
+            defaultMessage: 'Your profile data has been loaded',
+          })
+        );
+      },
+      onError() {
+        toggleNotification({
+          type: 'warning',
+          message: { id: 'notification.error', defaultMessage: 'An error occured' },
+        });
+      },
+    }
+  );
+
+  const { isLoading: isLoadingSSO, data: dataSSO } = useQuery(
+    ['providers', 'isSSOLocked'],
+    async () => {
+      const {
+        data: { data },
+      } = await get('/admin/providers/isSSOLocked');
```

Comment:
Should a type be inferred?

Reply:
yes, great reminding. I want to look at why i'm getting warnings for this, i keep missing them :(